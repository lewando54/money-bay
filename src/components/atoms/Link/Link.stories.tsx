/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import Link from './Link'

const meta = {
    component: Link,
    argTypes: {
        // To write
    }
} satisfies Meta<typeof Link>

export default meta
type Story = StoryObj<typeof Link>

export const Default: Story = {
    args: {
        // To write
    }
}
