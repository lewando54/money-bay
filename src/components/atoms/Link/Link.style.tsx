import { StyleSheet } from 'react-native'

import globalThemeStyle, { LINK_COLOR } from '../../../styling/GlobalTheme.style'
export default StyleSheet.create({
    text: {
        ...globalThemeStyle.text_Regular,
        fontSize: 16,
        color: LINK_COLOR
    }
})
