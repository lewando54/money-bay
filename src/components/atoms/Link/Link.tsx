import React from 'react'
import { StyleProp,  TextStyle } from 'react-native'

import { CustomText } from '@atoms'

import linkStyle from './Link.style'

interface ILinkProps {
    onPress?: () => void
    children?: string
    style?: StyleProp<TextStyle>
    testID?: string
}

export default function Link({
    onPress,
    children,
    style,
    testID
}: ILinkProps){
    return (
        <CustomText onPress={onPress} testID={testID} style={[linkStyle.text, style]}>{children}</CustomText>
    )
}
