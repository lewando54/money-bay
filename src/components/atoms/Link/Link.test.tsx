import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import Link from './Link'
describe('Link', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Link />)
    expect(tree).toMatchSnapshot()
  })
})
