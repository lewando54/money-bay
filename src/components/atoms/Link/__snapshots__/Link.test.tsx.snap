// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 14,
      },
      [
        {
          "color": "#FF5887",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
        },
        undefined,
      ],
    ]
  }
/>
`;
